Mining feature

Polyzonial opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->
Adds deposits, nuggets and deposit generation.
Adds Recipes to refine most of them into usable items

Modifies the mojave sun mining tool to do alot of extra damage to deposits without it becoming a god-slayer

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

Adds a mining system so we can get black lung

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Added ores, deposits and a generation system
add: Added crafting recipes via smelter
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Polyzonial committed:
Mining feature (DO NOT MERGE)

Adds deposits, nuggets and deposit generation.

Modifies the mojave sun mining tool to do alot of extra damage to deposits without it becoming a god-slayer

Polyzonial committed:
To be pushed to origin

Polyzonial committed:
forgot to tick ores.dm

Hehe hyup

Polyzonial committed:
Redoing and Polish

Redoes the ore mining process for better compatibility
mining_mult has been defined at a higher level as per hekzder's recommendations
ms_ores.dmi has been divided up to have inventory sprites work but putting the inventory sprites into materials_inventory.dmi
The ore deposit structures have been moved to the structure section of the mojave/icons folder

Polyzonial committed:
Ores don't suffocate you or spawn on occupied tiles

Does what it says

Polyzonial committed:
I forgot to make generation happen on Drought tiles

Hekzder commented:
DNMing this for now just because merging this so close to the test is probably not a great idea. We'll _potentially_ have it in for day two but can't guarantee. Right after the tests though this will def get fully reviewed and probably merged.

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+	desc = "Full of valuable errors to sell to coders!"
+	icon = ""
+	icon_state = ""
+	max_integrity = 400
```
It would be soul to have this random I think. Like ranges from 360-450? 

Not an insane range but enough to add some variance and have it so not all rocks are equal.

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+//deposits
+
+/obj/structure/ms13/ore_deposit/copper
+	name = "Copper ore deposit"
```
This applies to basically all the deposits but they don't need to be capitalized like this. I know it seems weird but for the sake of consistency that's just how it should be.

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+/obj/structure/ms13/ore_deposit/copper
+	name = "Copper ore deposit"
+	desc = "Full of copper ore, useful!"
+	icon = 'mojave/icons/structure/deposits.dmi'
```
This once again applies to basically all the deposits but you should just apply the icon path to the base ore_deposit so you don't have to define it every time. Once you do that you can remove the icon definition from each ore_deposit.

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+	icon = 'mojave/icons/structure/deposits.dmi'
+	icon_state = "copper-deposit"
+	density = FALSE
+	anchored = TRUE
```
Similar to above two comments, this applies to all deposits and is redundant. You can just define the density and anchored status on the base type ore_deposit.

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+
+/obj/structure/ms13/ore_deposit/alu
+	name = "Aluminum ore deposit"
+	desc = "Full of lead ore, lightweight!"
```
Typo

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+	deposit_type = /obj/item/stack/sheet/ms13/nugget_coal
+
+/obj/structure/ms13/ore_deposit/uranium
+	name = "Coal deposit"
```
Double typo here

Hekzder commented on mojave/structures/ores.dm:
```diff
@@ -0,0 +1,266 @@
+
+//Deposit generation
+
+#define DEPOSIT_SPAWN_LIST list(/obj/structure/ms13/ore_deposit/gold = 1, /obj/structure/ms13/ore_deposit/silver = 2, /obj/structure/ms13/ore_deposit/alu = 6, /obj/structure/ms13/ore_deposit/lead = 6, /obj/structure/ms13/ore_deposit/copper = 5)
```
You should add coal to this provided it's functioning properly and is all there. I assume maybe you originally left it out because you weren't able to find a use for it but I'll probably be able to use it for some things.

Hekzder commented on mojave/modules/crafting/recipes/recipes.dm:
```diff
@@ -247,3 +247,75 @@
+/datum/crafting_recipe/smelt_copper
+	name = "refine copper ore"
+	result = /obj/item/stack/sheet/ms13/refined_copper/two
+	time = 3 SECONDS
```
These should probably have a hammer tool requirement since you're presumably hammering out a couple ingots.

Polyzonial committed:
Changes and fixes

Randomises deposit health, lets coal spawn and fixes some types and corrects capitalisation.
Crafting ingots now requires a hammer.
Making steel requires coal.

Polyzonial committed:
I forgot the iron

:skull:

Polyzonial committed:
Vintage story update

ProfessorPopoff commented:
![image](https://github.com/Mojave-Sun/mojave-sun-13/assets/51188571/affeae1a-bea6-4241-91ed-089cada3ce62)
Ore spawns in on top of turfs with structures. This has to get fixed :DDD

Spawn chances are sussy but @Hekzder will do it. shoutout to @Hekzder

ProfessorPopoff commented:
Also you have compilation warnings for unused variables. Idk what you're using them for but you should compile and look at it. :)

mojave\structures\ores.dm:234:warning: turfDeposit: variable defined but not used
mojave\structures\ores.dm:246:warning: turfDeposit: variable defined but not used

Polyzonial committed:
More fixes, Remove redundant variable

Adds a better check, checks for both machines and structures before spawning in an ore deposit
Removes Redundant variable
adds an extra comment

Polyzonial committed:
Removes Duplicated comments

Oopsie!

Hekzder reviewed (approved):
This looks to be good now, merging this so I can make some adjustments before the dev test tomorrow. Good stuff man.
